samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -634,4 +651,20 @@ static ClusterBasedJobCoordinator createFromConfigLoader(Config submissionConfig
         metadataStore,
         config);
   }
+
+  @VisibleForTesting
+  static String[] toArgs(ApplicationConfig config) {
+    List<String> args = new ArrayList<>(config.size() * 2);
+
+    config.forEach((key, value) -> {
+      if (key.equals(ApplicationConfig.APP_MAIN_ARGS)) {
+        args.addAll(Arrays.asList(value.split("\\s")));
+      } else {
```

Comment:
is there else part being used right now?

Reply:
Yes,

if block is to pass native Beam pipeline options such as 
--runner SamzaRunner --maxSourceParallelism=1024

else block is to change other native samza configs with --config key=value, such as 

--config job.name=beam-test